test/remote-replication.test.js

```diff
@@ -0,0 +1,90 @@
+describe('Remote Replication', function() {
+  beforeEach(function(done) {
+    var test = this;
+
+    test.serverApp = loopback();
+    test.serverApp.use(function(req, res, next) {
+      console.log(req.method, req.url);
+      next();
+    });
+    test.serverApp.use(loopback.rest());
+    test.serverApp.enableAuth();
+    test.clientApp = loopback();
+    this.serverApp.set('legacyExplorer', false);
+    var settings = {
+      base: 'PersistedModel',
+      trackChanges: true,
+      dataSource: null,
+      http: {
+        path: 'my-model'
+      }
+    };
+    test.ServerModel = test.serverApp.model('ServerModel', settings);
+    test.ClientModel = test.clientApp.model('ClientModel', settings);
```

Comment:
No. This code is called before each test, and every time you call this code, a new model class is created and registered in the global static registry under the same name "ServerModel". It's safer to use a name that is unique *per test*, see replication tests for an example: https://github.com/strongloop/loopback/blob/131633f50d9d37f8f081bfe1f19005db92e44e99/test/replication.test.js#L21-L31

This is not strictly necessary, in many cases it's ok to override model with the same name. It's just that the case where it's not ok are difficult to identify, because the errors are very subtle.